extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -3,25 +3,114 @@
 
 import {
     RequestHeaders, Util, CorrelationIdHelper, TelemetryItemCreator, ICorrelationConfig,
-    RemoteDependencyData, DateTimeUtils, DisabledPropertyName, Data, IDependencyTelemetry,
-    IConfig, ConfigurationManager, ITelemetryContext, PropertiesPluginIdentifier, IPropertiesPlugin, DistributedTracingModes
+    RemoteDependencyData, DateTimeUtils, DisabledPropertyName, IDependencyTelemetry,
+    IConfig, ITelemetryContext, PropertiesPluginIdentifier, DistributedTracingModes
 } from '@microsoft/applicationinsights-common';
 import {
     CoreUtils, LoggingSeverity, _InternalMessageId,
     IAppInsightsCore, BaseTelemetryPlugin, ITelemetryPluginChain, IConfiguration, IPlugin, ITelemetryItem, IProcessTelemetryContext,
-    getLocation, getGlobal
+    getLocation, getGlobal, strUndefined, strPrototype, IInstrumentCallDetails, InstrumentFunc, InstrumentProto, getPerformance
 } from '@microsoft/applicationinsights-core-js';
-import { ajaxRecord } from './ajaxRecord';
+import { ajaxRecord, IAjaxRecordResponse } from './ajaxRecord';
 import { EventHelper } from './ajaxUtils';
 import { Traceparent } from './TraceParent';
+import dynamicProto from "@microsoft/dynamicproto-js";
```

Comment:
Major refactor to add XHR/Fetch window.performance additional metrics.
- Hiding private properties (by using dynamicProto)
- refactoring out duplicate code
Nett effect of this change on the 
Dependency plugin
- Unminified (browser) = additional code of just over 35kb
- Minified (browser) = 496 bytes LESS

AISKU
- Unminified (browser) = additional code of just under 35Kb
- Minified (browser) = 501 bytes LESS